SportsHub/SportsHub.Application/Services/UserService.cs

```diff
@@ -21,5 +22,11 @@ public UserService(IUnitOfWork unitOfWork)
         {
             return await unitOfWork.UserRepository.GetByUsernameAsync(username);
         }
+
+        public async Task SaveUserAsync(User user)
+        {
+            await unitOfWork.UserRepository.SaveUserAsync(user);
```

Comment:
i suppose here should be some validation, if user does not exist in the system for example

Reply:
I have validation inside the application layer. It checks if username or email already exists and if not it creates the user.